packages/tocco-ui/src/utilStyles/objectToCss.js

```diff
@@ -0,0 +1,35 @@
+import {theme} from 'styled-system'
+
+/**
+ * parse object as css, additionally request values from theme
+ * @param  {object} css declarations as object
+ *   simple --> {[css property as key]: [css property as string, integer or float]}
+ *   theme  --> {[css property as key]: [css property as array, first index =
+ *     theme path without indexes, subsequent indexes get retrieved]}
+ * @param  {object} react props
+ * @return {string} css
+ */
+
+const objectToCss = (declarations, props) => {
```

Comment:
here a solution with less string compositioning:

```
const objectToCss = (declarations, props) => {
  const css = []

  const getDefaultValue = themePath => theme(themePath, 0)(props)

  for (const [key, value] of Object.entries(declarations)) {
    let cssValue
    if (Array.isArray(value)) {
      const themePath = value.shift()
      if (value.length === 0) {
        cssValue = getDefaultValue(themePath)
      } else {
        const values = value.map(themeIndex => {
          const themeIdentifier = `${themePath}.${themeIndex}`
          return `${theme(themeIdentifier, 0)(props)}`
        })
        cssValue = values.join(' ')
      }
    } else {
      cssValue = value
    }

    css.push(`${key}: ${cssValue};`)
  }
  return css.join('\n') + '\n' // new line at end necessary? if no remove and adjust tests
}
```

Reply:
Awesome. Thank you for showing me your approach.

adopted in d37157758b1f1c32c0bf2188918f79a53685d6bb